frappe/desk/query_report.py

```diff
@@ -407,7 +407,8 @@ def build_xlsx_data(data, visible_idx, include_indentation, ignore_visible_idx=F
 						continue
 					label = column.get("label")
 					fieldname = column.get("fieldname")
-					cell_value = cstr(row.get(fieldname, row.get(label, "")))
+					cell_value = row.get(fieldname, row.get(label, ""))
+					cell_value = cstr(cell_value) if isinstance(cell_value, list) else cell_value
```

Comment:
We can't have any other non-primitive types right? Or should we err on side of caution and strigify any non-primitive types? 




reference:  https://openpyxl.readthedocs.io/en/stable/_modules/openpyxl/cell/cell.html





```python
TIME_TYPES = (datetime.datetime, datetime.date, datetime.time, datetime.timedelta)
TIME_FORMATS = {
    datetime.datetime:numbers.FORMAT_DATE_DATETIME,
    datetime.date:numbers.FORMAT_DATE_YYYYMMDD2,
    datetime.time:numbers.FORMAT_DATE_TIME6,
    datetime.timedelta:numbers.FORMAT_DATE_TIMEDELTA,
                }

STRING_TYPES = (str, bytes)
KNOWN_TYPES = NUMERIC_TYPES + TIME_TYPES + STRING_TYPES + (bool, type(None))
```

Reply:
Except for the Custom Cash flow report, I haven't come across a scenario where cell's have non-primitive data types and resulted in an exception being thrown.